R/providers.R

```diff
@@ -0,0 +1,8 @@
+#' Leaflet Providers.
```

Comment:
Let's rename this file to `data-providers.R` Just to signify it is for docs only.  

Same with `providers.details.R` to `data-providers.details.R`